internal/versioncheck/version.go

```diff
@@ -137,6 +137,18 @@ func CurrentGoFlutterTag(goDirectoryPath string) (currentTag string, err error)
 		return "", err
 	}
 
+	for _, pkg := range m.Replace {
+		if pkg.New.Path == expected {
+			return pkg.New.Version, nil
+		}
+	}
+
+	for _, pkg := range m.Require {
+		if pkg.Mod.Path == expected {
+			return pkg.Mod.Version, nil
+		}
+	}
+
 	if v := modx.Version(m, expected); v.Path == expected {
```

Comment:
Adding extra matching logic is a bit weird, that's acutally `modx.Version`'s task..

It seems like just dropping the `v.Path == expected` here would fix the problem. I'm just not sure why this match with `expected` was ever introduced. Perhaps we should only validate that v isn't empty. @james-lawrence thoughts?

Reply:
The added code was removed in @james-lawrence 's PR. I just re-added the old code.